src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -153,8 +153,9 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
         for dd in self.mgr.cache.get_daemons_by_service('snmp-gateway'):
             assert dd.hostname is not None
             assert dd.ports
+            addr = dd.ip if dd.ip else self._inventory_get_addr(dd.hostname)
             deps.append(dd.name())
-            snmp_gateway_urls.append(f"http://{dd.hostname}:{dd.ports[0]}/alerts")
+            snmp_gateway_urls.append(f"http://{addr}:{dd.ports[0]}/alerts")
```

Comment:
This becomes confusing to review. Can you amend the commits where those lines were introduced?